src/ethereum_test_tools/rpc/rpc.py

```diff
@@ -0,0 +1,118 @@
+"""
+JSON-RPC methods and helper functions for EEST consume based hive simulators.
+"""
+
+import time
+from typing import Dict, List, Literal, Union
+
+import requests
+from jwt import encode
+from tenacity import retry, stop_after_attempt, wait_exponential
+
+from ethereum_test_tools import Address
+
+
+class BaseRPC:
+    """
+    Represents a base RPC class for every RPC call used within EEST based hive simulators.
+    """
+
+    def __init__(self, client):
+        self.client = client
+        self.url = f"http://{client.ip}:8551"
+        self.jwt_secret = (
+            b"secretsecretsecretsecretsecretse"  # oh wow, guess its not a secret anymore
+        )
+
+    def generate_jwt_token(self):
+        """
+        Generates a JWT token based on the issued at timestamp and JWT secret.
+        """
+        iat = int(time.time())
+        return encode({"iat": iat}, self.jwt_secret, algorithm="HS256")
+
+    @retry(stop=stop_after_attempt(3), wait=wait_exponential(multiplier=1, max=10))
+    def post_request(self, method, params):
+        """
+        Sends a JSON-RPC POST request to the client RPC server at port 8551.
+        """
+        payload = {
+            "jsonrpc": "2.0",
+            "method": method,
+            "params": params,
+            "id": 1,
+        }
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {self.generate_jwt_token()}",
+        }
+
+        response = requests.post(self.url, json=payload, headers=headers)
+        response.raise_for_status()
+        result = response.json().get("result")
+
+        if result is None or "error" in result:
+            error_info = "result is None; and therefore contains no error info"
+            error_code = None
+            if result is not None:
+                error_info = result["error"]
+                error_code = error_info["code"]
+            raise Exception(
+                f"Error calling JSON RPC {method}, code: {error_code}, " f"message: {error_info}"
+            )
+
+        return result
+
+
+class EthRPC(BaseRPC):
+    """
+    Represents an `eth_X` RPC class for every default ethereum RPC method used within EEST based
+    hive simulators.
+    """
+
+    BlockNumberType = Union[int, Literal["latest", "earliest", "pending"]]
```

Comment:
I think I would move this type definition out of the class and expose it via __init__ for easier use in other modules.

Reply:
Added it to init here: [49352c5](https://github.com/ethereum/execution-spec-tests/pull/556/commits/49352c576871dc0c3978dc0901c81280ff5d6d8c)